core/src/main/scala/org/apache/spark/shuffle/ShuffleManager.scala

```diff
@@ -66,6 +66,18 @@ private[spark] trait ShuffleManager {
       context: TaskContext,
       metrics: ShuffleReadMetricsReporter): ShuffleReader[K, C]
 
+  /**
+   * Get a reader for the specific partitionIndex in map output statistics that are
+   * produced by range mappers. Called on executors by reduce tasks.
+   */
+  def getReaderForRangeMapper[K, C](
```

Comment:
we can probably merge this method with `getReaderForOneMapper`. one mapper is just a special range mappers.